### DISCUSSION TITLE
Does the finding post processing hook require exact string matches in it rules?

### DISCUSSION THREAD

**Discussion by moxli** [secureCodeBox/secureCodeBox#2110] [Q&A] [answered]
Hi,

I am scanning multiple endpoints with Nikto and ZAP which produce multiple findings I would love to manipulate.
These findings always include some ID which changes with every request, for example PHP Session IDs or trace IDs.
I want to manipulate these findings to trick DefectDojo into deduplicating these findings.
Because right now a new finding is created for each run due to deduplication not catching this.

For example Nikto finds a traceparent header:

```
id: 999100
msg: /: Uncommon header 'traceparent' found, with contents: 00-32f972bc67a26b32020b10b9c160a374-c2be9779c10862c1-01.
HTTP Method: GET
OSVDB: None

```

```
id: 999100
msg: /: Uncommon header 'traceparent' found, with contents: 00-ff46029edd56916bde91b93e46621f6d-c2be9779c10862c1-01.
HTTP Method: GET
OSVDB: None
```

Only the ID is different after each run.

![image](https://github.com/secureCodeBox/secureCodeBox/assets/26149616/5d514b56-66d2-49c9-a029-bd7bc80e7f67)

The ID is present in the name and description fields.

I thought about using the post processing hook to match for something like "Uncommon header 'traceparent' found, with contents" and then replace the entire line with "Uncommon header 'traceparent' found, with contents: REPLACED-FOR-DEDUPLICATION"?

Is it possible to use the post processing rules for this?

**Comment by J12934** [comment, marked as answer]
Hi @moxli 

> Does the finding post processing hook require exact string matches in it rules?

yes (unfortunatly) it does. The strings contain in the match pattern need to be a complete match for the finding post processing hook.

> Is it possible to use the post processing rules for this?

I also think the approach you are taking would unfortunatly not work as the DefectDojo hook is using the "raw" nikto results for the upload. So when you modify the findings it would not change the results in defectdojo because it is using the raw results.

**Comment by moxli** [comment]
In case anybody is interested:

I ended up creating my own Nikto image which just adds the headers I want to exclude to /nikto/databases/db_headers.

This prevents Nikto from reporting them in the first place :)
